lisp/Makefile

```diff
@@ -56,28 +52,31 @@ uninstall:
 	fi;
 
 clean:
-	rm -f $(ELC)
-	rm -f ess-autoloads.el
+	rm -f $(ELC) ess-autoloads.el julia-mode.el .dependencies
 
 distclean: clean
-# and potentially more
 
-### Targets below here are only for developers - and these must have perl
+
+### File Targets
+
+.dependencies: $(ELS)
```

Comment:
Why is this necessary?

Reply:
Auto-generates the pre-requisites for all our elc files. We don't need to hard-code them in our Makefile any longer.

Reply:
There is actually a more elegant way of doing that with .SECONDEXPANSION but it doesn't work on ancient make 3.8 on Travis' ubuntu trusty. So I gave up after a night of struggle.

Reply:
Sorry, I wasn't clear. I was thinking more along the lines of "why is it necessary to specify .elc dependencies in the first place?" If .elc doesn't exist yet, the byte compiler will just use the corresponding .el file. Can't we just say "all .elc files depend on the corresponding .el file" and not care about the order?

Reply:
But if one .el changes you need to recompile all els which depend on it. Or you wonder something else?

Reply:
Ah, I knew I was missing something obvious. I wasn't thinking about non-clean checkouts.